File: test/aidbox_sdk/snapshots/hl7-fhir-r4-core/MolecularSequence.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.Resource;

public class MolecularSequence : DomainResource, IResource
{
    public Base.ResourceReference? Patient { get; set; }
    public MolecularSequence_StructureVariant[]? StructureVariant { get; set; }
    public MolecularSequence_Repository[]? Repository { get; set; }
    public MolecularSequence_Variant[]? Variant { get; set; }
    public Base.ResourceReference? Specimen { get; set; }
    public string? Type { get; set; }
    public Base.ResourceReference[]? Pointer { get; set; }
    public string? ObservedSeq { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public MolecularSequence_Quality[]? Quality { get; set; }
    public Base.ResourceReference? Device { get; set; }
    public Base.Quantity? Quantity { get; set; }
    public required int CoordinateSystem { get; set; }
    public MolecularSequence_ReferenceSeq? ReferenceSeq { get; set; }
    public Base.ResourceReference? Performer { get; set; }
    public int? ReadCoverage { get; set; }

    public class MolecularSequence_StructureVariant_Inner : BackboneElement
    {
        public int? End { get; set; }
        public int? Start { get; set; }
    }

    public class MolecularSequence_StructureVariant_Outer : BackboneElement
    {
        public int? End { get; set; }
        public int? Start { get; set; }
    }

    public class MolecularSequence_StructureVariant : BackboneElement
    {
        public bool? Exact { get; set; }
        public MolecularSequence_StructureVariant_Inner? Inner { get; set; }
        public MolecularSequence_StructureVariant_Outer? Outer { get; set; }
        public int? Length { get; set; }
        public Base.CodeableConcept? VariantType { get; set; }
    }

    public class MolecularSequence_Repository : BackboneElement
    {
        public string? Url { get; set; }
        public string? Name { get; set; }
        public required string Type { get; set; }
        public string? DatasetId { get; set; }
        public string? ReadsetId { get; set; }
        public string? VariantsetId { get; set; }
    }

    public class MolecularSequence_Variant : BackboneElement
    {
        public int? End { get; set; }
        public string? Cigar { get; set; }
        public int? Start { get; set; }
        public string? ObservedAllele { get; set; }
        public Base.ResourceReference? VariantPointer { get; set; }
        public string? ReferenceAllele { get; set; }
    }

    public class MolecularSequence_Quality_Roc : BackboneElement
    {
        public int[]? NumFN { get; set; }
        public int[]? NumFP { get; set; }
        public int[]? NumTP { get; set; }
        public int[]? Score { get; set; }
        public string[]? FMeasure { get; set; }
        public string[]? Precision { get; set; }
        public string[]? Sensitivity { get; set; }
    }

    public class MolecularSequence_Quality : BackboneElement
    {
        public string? TruthTP { get; set; }
        public string? FScore { get; set; }
        public string? TruthFN { get; set; }
        public string? QueryFP { get; set; }
        public Base.CodeableConcept? Method { get; set; }
        public string? Precision { get; set; }
        public int? Start { get; set; }
        public string? QueryTP { get; set; }
        public required string Type { get; set; }
        public string? Recall { get; set; }
        public MolecularSequence_Quality_Roc? Roc { get; set; }
        public Base.Quantity? Score { get; set; }
        public int? End { get; set; }
        public Base.CodeableConcept? StandardSequence { get; set; }
        public string? GtFP { get; set; }
    }

    public class MolecularSequence_ReferenceSeq : BackboneElement
    {
        public Base.CodeableConcept? Chromosome { get; set; }
        public Base.CodeableConcept? ReferenceSeqId { get; set; }
        public int? WindowEnd { get; set; }
        public string? Strand { get; set; }
        public string? GenomeBuild { get; set; }
        public string? Orientation { get; set; }
        public Base.ResourceReference? ReferenceSeqPointer { get; set; }
        public string? ReferenceSeqString { get; set; }
        public int? WindowStart { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/MolecularSequence.py
from typing import Optional, List
from pydantic import *
from ..base.CodeableConcept import CodeableConcept
from ..base.Quantity import Quantity
from ..base.DomainResource import DomainResource
from ..base.Reference import Reference
from ..base.Identifier import Identifier
from ..base.BackboneElement import BackboneElement

class MolecularSequence_StructureVariant_Inner(BackboneElement):
    end: Optional[int] = None
    start: Optional[int] = None

class MolecularSequence_StructureVariant_Outer(BackboneElement):
    end: Optional[int] = None
    start: Optional[int] = None

class MolecularSequence_StructureVariant(BackboneElement):
    exact: Optional[bool] = None
    inner: Optional[MolecularSequence_StructureVariant_Inner] = None
    outer: Optional[MolecularSequence_StructureVariant_Outer] = None
    length: Optional[int] = None
    variant_type: Optional[CodeableConcept] = None

class MolecularSequence_Repository(BackboneElement):
    url: Optional[str] = None
    name: Optional[str] = None
    type: str
    dataset_id: Optional[str] = None
    readset_id: Optional[str] = None
    variantset_id: Optional[str] = None

class MolecularSequence_Variant(BackboneElement):
    end: Optional[int] = None
    cigar: Optional[str] = None
    start: Optional[int] = None
    observed_allele: Optional[str] = None
    variant_pointer: Optional[Reference] = None
    reference_allele: Optional[str] = None

class MolecularSequence_Quality_Roc(BackboneElement):
    num_f_n: Optional[List[int]] = None
    num_f_p: Optional[List[int]] = None
    num_t_p: Optional[List[int]] = None
    score: Optional[List[int]] = None
    f_measure: Optional[List[float]] = None
    precision: Optional[List[float]] = None
    sensitivity: Optional[List[float]] = None

class MolecularSequence_Quality(BackboneElement):
    truth_t_p: Optional[float] = None
    f_score: Optional[float] = None
    truth_f_n: Optional[float] = None
    query_f_p: Optional[float] = None
    method: Optional[CodeableConcept] = None
    precision: Optional[float] = None
    start: Optional[int] = None
    query_t_p: Optional[float] = None
    type: str
    recall: Optional[float] = None
    roc: Optional[MolecularSequence_Quality_Roc] = None
    score: Optional[Quantity] = None
    end: Optional[int] = None
    standard_sequence: Optional[CodeableConcept] = None
    gt_f_p: Optional[float] = None

class MolecularSequence_ReferenceSeq(BackboneElement):
    chromosome: Optional[CodeableConcept] = None
    reference_seq_id: Optional[CodeableConcept] = None
    window_end: Optional[int] = None
    strand: Optional[str] = None
    genome_build: Optional[str] = None
    orientation: Optional[str] = None
    reference_seq_pointer: Optional[Reference] = None
    reference_seq_string: Optional[str] = None
    window_start: Optional[int] = None

class MolecularSequence(DomainResource):
    patient: Optional[Reference] = None
    structure_variant: Optional[List[MolecularSequence_StructureVariant]] = None
    repository: Optional[List[MolecularSequence_Repository]] = None
    variant: Optional[List[MolecularSequence_Variant]] = None
    specimen: Optional[Reference] = None
    type: Optional[str] = None
    pointer: Optional[List[Reference]] = None
    observed_seq: Optional[str] = None
    identifier: Optional[List[Identifier]] = None
    quality: Optional[List[MolecularSequence_Quality]] = None
    device: Optional[Reference] = None
    quantity: Optional[Quantity] = None
    coordinate_system: int
    reference_seq: Optional[MolecularSequence_ReferenceSeq] = None
    performer: Optional[Reference] = None
    read_coverage: Optional[int] = None